app/api/v1/endpoints/strategies.py

```
@@ -1069,6 +1069,390 @@ async def get_publisher_stats(
         )
 
 
+@router.get("/publisher/earnings-history")
+async def get_publisher_earnings_history(
+    period: str = "30d",
+    current_user: User = Depends(get_current_user),
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)